snowflake/datadog_checks/snowflake/config_models/validators.py

```diff
@@ -8,4 +8,10 @@ def initialize_instance(values, **kwargs):
     if 'username' not in values and 'user' in values:
         values['username'] = values['user']
 
+    if 'private_key_password' in values and 'private_key_path' not in values:
```

Comment:
Can these values be used along username and password?

Reply:
These values shouldn't be used alongside `password`
I'll open a new PR after this one is merged to refactor the whole config validation ; since I'm adding other authentication methods at the same time